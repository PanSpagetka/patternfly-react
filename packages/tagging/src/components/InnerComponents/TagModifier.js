import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Form } from 'patternfly-react';

const TagModifier = ({ header, children }) => (
  <React.Fragment>
    <Row><Col lg={12}><h2>{header}</h2></Col></Row>
    <Form horizontal>
      {children}
    </Form>

  </React.Fragment>
);

TagModifier.propTypes = {
  header: PropTypes.string,
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
};

TagModifier.defaultProps = {
  header: 'Add/Modify tag',
};

export default TagModifier;
