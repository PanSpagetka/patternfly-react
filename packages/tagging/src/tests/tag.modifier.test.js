import React from 'react';
import TagModifier from '../components/TagModifier';
import renderer from 'react-test-renderer';

const tags = [
  {
    description: 'Name',
    id: 1,
    values: [{ description: 'Pepa', id: 11 }, { description: 'Franta', id: 12 }]
  },
  {
    description: 'Number',
    id: 2,
    values: [{ description: '1', id: 21 }, { description: '2', id: 22 }]
  },
  {
    description: 'Animal',
    id: 3,
    values: [
      { description: 'Duck', id: 31 },
      { description: 'Cat', id: 32 },
      { description: 'Dog', id: 33 }
    ]
  },
  {
    description: 'Food',
    id: 4,
    values: [
      { description: 'Steak', id: 41 },
      { description: 'Duck', id: 42 },
      { description: 'Salad', id: 43 }
    ]
  },
  {
    description: 'Something',
    id: 5,
    values: [
      { description: 'Knedlik', id: 51 },
      {
        description:
          'Daenerys Stormborn of the House Targaryen, First of Her Name,...and Mother of Dragons',
        id: 52
      }
    ]
  }
];

const selectedTagCategory = { description: 'animal', id: 1 };
const selectedTagValue = { description: 'duck', id: 1 };

function onChange(x) {
  return x;
}
describe('Tagging modifier', () => {
  it('match snapshot', () => {
    const component = renderer.create(
      <TagModifier
        tags={tags}
        onTagValueChange={onChange}
        onTagCategoryChange={onChange}
        selectedTagCategory={selectedTagCategory}
        selectedTagValue={selectedTagValue}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
