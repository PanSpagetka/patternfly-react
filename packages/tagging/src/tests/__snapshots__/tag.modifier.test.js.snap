// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tagging modifier match snapshot 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <h2>
        Add/Modify tag
      </h2>
    </div>
  </div>,
  <form
    className="form-horizontal"
  />,
]
`;
